agent/agent.go

```diff
@@ -235,7 +235,9 @@ func (a *Agent) loop(ctx context.Context) error {
 	timer := time.NewTimer(0)
 	defer timer.Stop()
 
-	tc := &taskContext{}
+	tc := &taskContext{
+		logger: client.NewSingleChannelLogHarness("agent.error", a.defaultLogger),
```

Comment:
nit: this might not be for logging errors necessarily, just that there's no other logger available to use. Could we call this `"default'` or something instead of `"agent.error"`? Same for the others.